.github/workflows/approved-for-ci-run.yml

```diff
@@ -90,13 +90,47 @@ jobs:
 
           ALREADY_CREATED="$(gh pr --repo ${GITHUB_REPOSITORY} list --head ${BRANCH} --base main --json number --jq '.[].number')"
           if [ -z "${ALREADY_CREATED}" ]; then
+            LABELS=$(gh pr view ${PR_NUMBER}  --json labels --jq '.labels.[].name' | \
+            awk 'BEGIN {labels="run-e2e-tests-in-draft"} {if ($1 ~ /^run/) labels=labels "," $1;} END {print labels}')
             gh pr --repo "${GITHUB_REPOSITORY}" create --title "CI run for PR #${PR_NUMBER}" \
                                                        --body-file "body.md" \
                                                        --head "${BRANCH}" \
                                                        --base "main" \
-                                                       --label "run-e2e-tests-in-draft" \
+                                                       --label $LABELS \
                                                        --draft
+          else
+            echo "ALREADY_CREATED=$ALREADY_CREATED" >> $GITHUB_ENV
+            for OLD_PR in $ALREADY_CREATED
+            do
+              LABELS_TO_REMOVE=$(comm -23 <(gh pr --repo ${GITHUB_REPOSITORY} view $OLD_PR --json labels --jq '.labels.[].name'| grep -E '^run' | sort) <(gh pr --repo ${GITHUB_REPOSITORY} view $PR_NUMBER --json labels --jq '.labels.[].name' |  grep -E '^run' | sort ))
+              LABELS_TO_REMOVE=${LABELS_TO_REMOVE//run-e2e-tests-in-draft/}
+              LABELS_TO_REMOVE=$(echo $LABELS_TO_REMOVE | awk 'BEGIN {list=""} {for (i=1;i<=NF;i++) if (length($i)>0) if (length(list)>0) list=list "," $i; else list=$i;} END {print list}')
+              LABELS_TO_ADD=$(comm -13 <(gh pr --repo ${GITHUB_REPOSITORY} view $OLD_PR --json labels --jq '.labels.[].name'|  grep -E '^run' |sort) <(gh pr --repo ${GITHUB_REPOSITORY} view $PR_NUMBER --json labels --jq '.labels.[].name' |  grep -E '^run' | sort ))
+              LABELS_TO_ADD=$(echo $LABELS_TO_ADD | awk 'BEGIN {list=""} {for (i=1;i<=NF;i++) if (length($i)>0) if (length(list)>0) list=list "," $i; else list=$i;} END {print list}')
+              echo "LABELS_TO_ADD_${OLD_PR}=$LABELS_TO_ADD" >> $GITHUB_ENV
+              echo "LABELS_TO_REMOVE_${OLD_PR}=$LABELS_TO_REMOVE" >> $GITHUB_ENV
```

Comment:
I'd use step outputs instead of setting it to global env.
You can check the usage of `GITHUB_OUTPUT` as an example